### ISSUE TITLE
Support for savepoints

### ISSUE THREAD

**Issue by gigatexal** [vitessio/vitess#4462]
#### Overview of the Issue

The support for savepoints would be nice even for use within a single shard. 

#### Operating system and Environment details

Alpine Linux, Docker, Python 2.7, SqlAlchemy

#### Log Fragments

```
InternalError: (pymysql.err.InternalError) (1105, u'vtgate: http://vtgate-zone1-7c9f75979b-fbv2f:15001/: unrecognized statement: SAVEPOINT sa_savepoint_1') [SQL: u'SAVEPOINT sa_savepoint_1'] (Background on this error at: http://sqlalche.me/e/2j85)
```

**Comment by georgeclaghorn** [comment]
ORMs like Ruby on Rails’s [Active Record](https://guides.rubyonrails.org/active_record_basics.html) often rely on savepoints to implement nested transactions. Active Record in particular doesn’t currently account for DBMSes that *don’t* implement savepoints, since the ones it natively supports (MySQL, PostgreSQL, sqlite3) all do.

**Comment by hallaroo** [comment]
Work on this feature was addressed by PR [#6438](https://github.com/vitessio/vitess/pull/6438). Closing out. Thanks!

**Comment by deepthi** [comment]
Fixed by #6404 #6414 #6438
